src/games/strategy/triplea/ui/TripleaMenu.java

```diff
@@ -407,21 +408,25 @@ public void actionPerformed(final ActionEvent e) {
   private void addUnitNationDrawMenu(final JMenu parentMenu){
     final JMenu unitSizeMenu = new JMenu();
     unitSizeMenu.setMnemonic(KeyEvent.VK_N);
-    unitSizeMenu.setText("Set Flag Display Mode");
+    unitSizeMenu.setText("Flag Display Mode");
     
-    final ButtonGroup unitNationGroup = new ButtonGroup();
-    unitSizeMenu.add(createRadioButtonItem("Hide Nation", unitNationGroup, SwingAction.of(e -> {
-      UnitsDrawer.setUnitFlagDrawMode(UnitsDrawer.UnitFlag.NONE);
-      frame.getMapPanel().resetMap();
-      }), false));
-    unitSizeMenu.add(createRadioButtonItem("Next to Unit", unitNationGroup, SwingAction.of(e -> {
-      UnitsDrawer.setUnitFlagDrawMode(UnitsDrawer.UnitFlag.NEXT_TO);
-      frame.getMapPanel().resetMap();
-      }), true));
-    unitSizeMenu.add(createRadioButtonItem("Below Unit", unitNationGroup, SwingAction.of(e -> {
-      UnitsDrawer.setUnitFlagDrawMode(UnitsDrawer.UnitFlag.BELOW);
-      frame.getMapPanel().resetMap();
-      }), false));
+    Preferences prefs = Preferences.userNodeForPackage(getClass());
+    String setting = prefs.get("UNIT_FLAG_DRAW_MODE", UnitsDrawer.UnitFlagDrawMode.NEXT_TO.toString());
+    
+    if(UnitsDrawer.UnitFlagDrawMode.NONE.toString().equals(setting)){
+      UnitsDrawer.setUnitFlagDrawMode(UnitsDrawer.UnitFlagDrawMode.NONE, prefs);
+    }
+    if(UnitsDrawer.UnitFlagDrawMode.NEXT_TO.toString().equals(setting)){
+      UnitsDrawer.setUnitFlagDrawMode(UnitsDrawer.UnitFlagDrawMode.NEXT_TO, prefs);    
+    }
+    if(UnitsDrawer.UnitFlagDrawMode.BELOW.toString().equals(setting)){
+      UnitsDrawer.setUnitFlagDrawMode(UnitsDrawer.UnitFlagDrawMode.BELOW, prefs);
+    }
+    
+    final ButtonGroup unitFlagSettingGroup = new ButtonGroup();
+    unitSizeMenu.add(createFlagDrawModeRadionButtonItem("Hide Nation", unitFlagSettingGroup, UnitsDrawer.UnitFlagDrawMode.NONE, setting, prefs));
```

Comment:
I recommend for menu option names: "None" "Small" "Large"
As is, they do not seem very consistent, and "Hide Nation" sounds a bit like all units would be hidden. 
The text is user facing, so kindly spending a bit of extra time on it to get it right is valuable.